core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1787,6 +1792,12 @@ private[spark] class DAGScheduler(
       case _: ExceptionFailure | _: TaskKilled => updateAccumulators(event)
       case _ =>
     }
+    if (trackingCacheVisibility) {
+      // Update rdd blocks' visibility status.
+      blockManagerMaster.updateRDDBlockVisibility(
```

Comment:
An asynchronous call seems enough here?

Reply:
Yes, async call should be fine. Will make the change, thanks.

Reply:
Updated